New entry cicd (#1752)
Codecademy/docs: content/general/concepts/cicd-pipeline/cicd-pipeline.md

---
Title: 'CICD Pipeline'
Description: 'The CICD pipeline automates the software/application delivery process.'
Subjects: 
  - 'Developer Tools'
  - 'Mobile Development'
  - 'Web Design'
  - 'Web Development'
Tags:
  - 'Developer Tools'
  - 'Development'
CatalogContent:

>   - 'paths/front-end-engineer-career-path'

Author: Hello! Are you referring to the Title or a larger change to the Metadata?

  - 'paths/back-end-engineer-career-path'
---

> The **CICD Pipeline** is the combination of continuous integration, delivery and deployment of software development and application deployment. **CI** stands for **Continuous Integration**. This is the automation process for Developers. **CD** stands for both **Continous Delivery** and **Continuous Deployment**. Continuous Delivery is the automated delivery of code to the testing and development steps of the pipeline. These environments are usually in a repository like [Github](https://www.github.com). Continuous Deployment refers to code passing automated tests and then automatically placed in production, which in turn is used for automatic deployment. The four main steps of the pipeline are: 

Reviewer: ```suggestion
The **CICD Pipeline** is the combination of continuous integration, delivery, and deployment of software development and application deployment.

CI (Continuous Integration) is the automation process for Developers. CD (Continuous Delivery and Continuous Deployment) is the automated delivery of code to the testing and development steps of the pipeline. These environments are usually in a repository like Github. Continuous deployment refers to code passing automated tests and then placed in production, which in turn is used for automatic deployment.
```

> 1. **Source** - The creation of code by developers, which is continuously integrated within a shared repository. Version control systems, like [Git](https://git-scm.com/) are used for tracking and managing changes to the software. When changes are commited to the repository, automated builds and testing are triggered.

Reviewer: ```suggestion
## Four Steps in CICD Pipeline

1. Source: The creation of code by developers, which is continuously integrated within a shared repository. Version control systems, like Git are used for tracking and managing changes to the software. When changes are commited to the repository, automated builds and tests are triggered.
```

> 2. **Build** - The process of using code from the repository to link necessary dependencies and libraries to "build" an executable file. Scripts can be utilized to convert the file in to a Virtual Machine or Container. Logs generated during this time are used to find errors and send notifications to developers when the build is complete. 
> 
> 3. **Test** - Automated testing tools confirm the build is working and any new additons are not causing disruptions to a previous working version. If there are any errors, the test results are sent back to the developers to fix. 
> 
> 4. **Deploy** - After a build sucessfully passes rigorous testing, it is automatically deployed to the production environment. Since no deployment is perfect, there are still ways to report the unexpected errors and alert developers, which starts the process all over again. 

Reviewer: ```suggestion
2. Build: The process of using code from the repository to link necessary dependencies and libraries to "build" an executable file. Scripts can be utilized to convert the file in to a Virtual Machine or Container. Logs generated during this time are used to find errors and send notifications to developers when the build is complete. 

3. Test: Automated testing tools confirm the build is working and any new additons are not causing disruptions to a previous working version. If there are any errors, the test results are sent back to the developers to fix. 

4. Deploy: After a build sucessfully passes rigorous testing, it is automatically deployed to the production environment. Since no deployment is perfect, there are still ways to report the unexpected errors and alert developers, which starts the process all over again. 
```

---
Title: 'CICD Pipeline'
Description: 'The CICD pipeline automates the software/application delivery process.'
Subjects: 
  - 'Developer Tools'
  - 'Mobile Development'
  - 'Web Design'
  - 'Web Development'
Tags:
  - 'Developer Tools'
  - 'Development'
CatalogContent:
  - 'paths/front-end-engineer-career-path'
  - 'paths/back-end-engineer-career-path'
---

The **CICD Pipeline** is the combination of continuous integration, delivery, and deployment of software development and application deployment.

> CI (Continuous Integration) is the automation process for Developers. CD (Continuous Delivery and Continuous Deployment) is the automated delivery of code to the testing and development steps of the pipeline. These environments are usually in a repository like Github. Continuous deployment refers to code passing automated tests and then placed in production, which in turn is used for automatic deployment.

Reviewer 2: ```suggestion
CI (Continuous Integration) is the automation process for developers. CD (Continuous Delivery and Continuous Deployment) is the automated delivery of code to the testing and development steps of the pipeline. These environments are usually in a repository like Github. Continuous deployment refers to code passing automated tests and then placed in production, which in turn is used for automatic deployment.
```

> ## Four Steps in CICD Pipeline

Reviewer 2: ```suggestion
## The Four Phases of a CICD Pipeline
```

> - **The other CD**: The second CD stands for continuous deployment. When code passes the automated tests, it is automatically placed in production, which is then used for automatic deployment. 

Reviewer: It would be good to include this in the introductory paragraph above. This way, we get the meaning of CICD after we mention it.

Since the intro ends with going into the steps of the pipeline, after the colon, you can turn that into a numbered list like:

1. Source
2. Build
... etc

## Steps of the CICD Pipeline

There are four main steps to the CICD Pipeline: 
1. Source
2. Build
3. Test

> 4. Deploy

Reviewer: Rather than just listing these out, let's go ahead and define them in your own words of course. I recommend looking into: https://circleci.com/docs/about-circleci/

> 2. Build: The process of using code from the repository to link necessary dependencies and libraries to "build" an executable file. Scripts can be utilized to convert the file in to a Virtual Machine or Container. Logs generated during this time are used to find errors and send notifications to developers when the build is complete. 

Reviewer 2: ```suggestion
2. Build: The process of using code from the repository to link necessary dependencies and libraries to "build" an executable file. Scripts can be utilized to convert the file to a [virtual machine](https://www.codecademy.com/resources/docs/general/virtual-machines) or container. Logs generated during this time are used to find errors and send notifications to developers when the build is complete. 
```

Author: Thank you for helping me make this a better document and finding the errors that I missed! All should be updated now!

> 3. Test: Automated testing tools confirm the build is working and any new additons are not causing disruptions to a previous working version. If there are any errors, the test results are sent back to the developers to fix. 

Reviewer 2: ```suggestion
3. Test: Automated testing tools confirm the build is working and any new additions are not causing disruptions to a previous working version. If there are any errors, the test results are sent back to the developers to fix. 
```
